RecoLocalTracker/SiPixelRecHits/python/SiPixelRecHits_cfi.py

```diff
@@ -6,6 +6,39 @@
     VerboseLevel = cms.untracked.int32(0)
 )
 
-siPixelRecHitsPreSplitting = siPixelRecHits.clone(
+_siPixelRecHitsPreSplitting = siPixelRecHits.clone(
     src = 'siPixelClustersPreSplitting'
 )
+
+from HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA
+siPixelRecHitsPreSplitting = SwitchProducerCUDA(
+    cpu = _siPixelRecHitsPreSplitting.clone()
+)
+
+
+
+from Configuration.ProcessModifiers.gpu_cff import gpu
+from RecoLocalTracker.SiPixelRecHits.siPixelRecHitCUDA_cfi import siPixelRecHitCUDA as _siPixelRecHitCUDA
+from RecoLocalTracker.SiPixelRecHits.siPixelRecHitFromSOA_cfi import siPixelRecHitFromSOA as _siPixelRecHitFromSOA
+
+gpu.toModify(siPixelRecHitsPreSplitting, 
+    cuda = _siPixelRecHitFromSOA.clone()
+)
+
+
+siPixelRecHitsPreSplittingTask = cms.Task(siPixelRecHitsPreSplitting)
+
+siPixelRecHitsCUDAPreSplitting = _siPixelRecHitCUDA.clone(
+    beamSpot = "offlineBeamSpotToCUDA"
+)
+
+siPixelRecHitsLegacyPreSplitting = _siPixelRecHitFromSOA.clone()
+siPixelRecHitsPreSplittingTaskCUDA = cms.Task(
+    siPixelRecHitsCUDAPreSplitting,
+    siPixelRecHitsLegacyPreSplitting,
```

Comment:
```suggestion
    siPixelRecHitsPreSplittingCUDA,
    siPixelRecHitsyPreSplittingLegacy,
```
would seem to follow the style more clearly

Reply:
Done in https://github.com/cms-patatrack/cmssw/pull/593/commits/dc20554b972d0180339acf6f4f6a8536f0c81d1c